DonjinKrawler-Client/src/main/java/donjinkrawler/Game.java

```diff
@@ -10,34 +10,52 @@
 
 public class Game extends JPanel implements ActionListener {
 
+    enum DoorDirection {
+        LEFT,
+        TOP,
+        RIGHT,
+        BOTTOM
+    }
+
     private final Timer timer;
     private final Player player;
-    private final GameMap gameMap;
+    private GameMap gameMap;
     private final int delay = 10;
     private int mouseX = 0;
     private int mouseY = 0;
     private double angle;
     private double rotate;
     private int counter = 0;
     private static final Set<PlayerShell> shells = new HashSet<>();
+    private int[][] mapGrid;
```

Comment:
Dabartinė implementacija kinda invalidates klasių diagramą. Pagal ją tai GameMap turėtų turėt GameLevel, kur anie turi elementus or something. Aišku visa tai buvo theoretical, bet gali kliūti, kad klasių diagramoj vienaip, o pas mus kode kitaip. 
Pirmiausia turbūt lygiai turėtų būti saugomi GameMap klasėje, o ne Game klasėje. Kai plėsim lygius, Game klasė nebus tokia išsipūtusi ir jai nelabai aktualu patys lygiai, gal labiau tik Map'as.
Vietoj to, kad saugoti žemėlapį dvimačiame int'ų masyve, galima pakurti GameLevel klasę, kuri saugotų savo state (ar cleared ar ne), savo skaičių, esančius elementus ir etc. ir tuo pačiu saugotų kitų levelių references kaip North, South, East, West door. Tada ir GameMap užtektų vienos klasės - rootLevel. Bent man tai atrodo patraukliau nei žaisti su dvimačiu masyvu, bet kadangi per daug į esamą implementaciją nesigilinau, tai nežinau kaip lengvai tai įgyvendinama būtų client / server komunikacijos kontekste. Just food for thought and let me know what you think.

Reply:
Tai jo, realizacija tikrai gali būti geresnė. Dabar kol kodo nežiūri, gal nekeisiu, kad Arvydui nesikirstų dalykai, savaitgalį pabandysiu pritempti prie diagramos viską. Su klasėm žinoma viskas būtų daug gražiau, bet nežinau kaip sinchronizuoti viską reikėtų. Kai visi žaidėjai turi po 2d celių kopiją tada galima forcint visus prie mapGrid[x][y] celės. O su klasėm turbūt gautusi taip, kad jei du žaidėjai identiškai išeina pro skirtingas duris servas pirma išsiųstų "Eikit į kairijį kambarį", poto "eikit į dešinį kambarį" ir gautųsi kad visi nušoka arba per toli arba išvis skirtingai. 
Reikia nebent apgalvoti sinchronizavimo galimybes geresnes.